Core/include/Acts/TrackFitting/KalmanFitter.hpp

```diff
@@ -646,32 +651,31 @@ class KalmanFitter {
         // the lastTrackIndex.
         result.lastMeasurementIndex = result.lastTrackIndex;
 
-      } else if (surface->associatedDetectorElement() != nullptr ||
-                 surface->surfaceMaterial() != nullptr) {
+      } else if ((precedingMeasurementExists && surfaceIsSensitive) ||
+                 surfaceHasMaterial) {
         // We only create track states here if there is already measurement
         // detected or if the surface has material (no holes before the first
         // measurement)
-        if (result.measurementStates > 0 ||
-            surface->surfaceMaterial() != nullptr) {
-          auto trackStateProxyRes = detail::kalmanHandleNoMeasurement(
-              state, stepper, *surface, *result.fittedStates,
-              result.lastTrackIndex, true, logger(), freeToBoundCorrection);
-
-          if (!trackStateProxyRes.ok()) {
-            return trackStateProxyRes.error();
-          }
+        auto trackStateProxyRes = detail::kalmanHandleNoMeasurement(
+            state, stepper, *surface, *result.fittedStates,
+            result.lastTrackIndex, true, logger(), precedingMeasurementExists,
+            freeToBoundCorrection);
 
-          const auto& trackStateProxy = *trackStateProxyRes;
-          result.lastTrackIndex = trackStateProxy.index();
+        if (!trackStateProxyRes.ok()) {
+          return trackStateProxyRes.error();
+        }
 
-          if (trackStateProxy.typeFlags().test(TrackStateFlag::HoleFlag)) {
-            // Count the missed surface
-            result.missedActiveSurfaces.push_back(surface);
-          }
+        const auto& trackStateProxy = *trackStateProxyRes;
+        result.lastTrackIndex = trackStateProxy.index();
 
-          ++result.processedStates;
+        if (trackStateProxy.typeFlags().test(TrackStateFlag::HoleFlag)) {
+          // Count the missed surface
+          result.missedActiveSurfaces.push_back(surface);
         }
-        if (surface->surfaceMaterial() != nullptr) {
+
+        ++result.processedStates;
+
+        if (surfaceHasMaterial) {
```

Comment:
I think we could drop this if since this is also checked inside the function

Reply:
Should be fine. We get an extra line of VERBOSE output for sensitive material-less surfaces, but they don't seem to be confusing.

Reply:
Sure this does not hurt but it is inconsistent with how the `materialInteractor` is called above eg
https://github.com/acts-project/acts/blob/4cf5f3fa5420f4c103097fab1dcbff91acdb0b80/Core/include/Acts/TrackFitting/KalmanFitter.hpp#L608-L610